src/pybind/mgr/cephadm/module.py

```diff
@@ -88,6 +89,19 @@ def _name_to_entity_name(name):
     else:
         return name
 
+def is_invalid_host(name):
+    p = re.compile('^[a-zA-Z0-9-]+$')
+    if len(name) > 250:
+        return 'name is too long (max 250 chars)'
+    parts = name.split('.')
+    for part in name.split('.'):
+        if len(part) == 0:
+            return '.-delimited name component must not be empty'
+        if len(part) > 63:
+            return '.-delimited name component must not be more than 63 chars'
+        if not p.match(part):
+            return 'name component must include only a-z, 0-9, and -'
+    return None
```

Comment:
What about using 'assertive' statements rather than defensive ones?
```py
def assert_valid_host(host):
  try:
    assert len(name) <= 250, 'name is too long (max 250 chars)'
    assert ...
  except AssertionError as e:
    raise OrchestratorError(e)
...

def add_host(self, host):
  assert_valid_host(host)
  out, err, code = self._run_cephadm(host, 'client', 'check-host',...
...
```

Reply:
that's way nicer! updated